erpnext/public/js/utils.js

```diff
@@ -514,7 +514,32 @@ $.extend(erpnext.utils, {
 				</div>
 			`;
 		}
+	},
+
+    setup_last_billed_rate_formatter(doctype, fieldname) {
+		let df = frappe.meta.get_docfield(doctype, fieldname);
+		if (df) {
+			erpnext.utils.set_last_billed_rate_link_formatter(df);
+		}
+	},
+	
+	set_last_billed_rate_link_formatter(df) {
+		df.formatter = (value, df, options, doc) => {
+			if (!value) return "";
+	
+			if (doc?.item_code && cur_frm?.doc?.customer && cur_frm?.doc?.company) {
+				const customer = cur_frm.doc.customer;
+				const company = cur_frm.doc.company;
+	
+				const link = `/app/query-report/Sales Details?company=${encodeURIComponent(company)}&customer=${encodeURIComponent(customer)}&item_code=${encodeURIComponent(doc.item_code)}`;
+	
+				return `<a href="${link}" target="_blank" style="text-decoration: underline;">${value}</a>`;
```

Comment:
- no need for text-decoration
- use standard formatter before printing value, are your printing unformatted value
- see for reference: https://github.com/ParaLogicTech/erpnext/blob/7e83855084f8fbcc84391da9a58d8cb664b821a7/erpnext/public/js/utils.js#L624-L640